guides/source/upgrading_ruby_on_rails.md

```diff
@@ -319,7 +319,7 @@ By opting-out you optimize `$LOAD_PATH` lookups (less directories to check), and
 
 #### Thread-safety
 
-In classic mode constant autoloading is not thread-safe, though Rails has locks in place for example to make web requests thread-safe when autoloading is enabled, as it is common in `development` mode.
+Constant autoloading is not thread-safe in classic mode, though Rails has locks in place for example to make web requests thread-safe when autoloading is enabled, as it is common in `development` mode.
```

Comment:
The `In classic mode constant autoloading is not thread-safe` sentence is a bit confusing.

There are two ways to clear this confusion.

1. Use comma like

```diff
- In classic mode constant autoloading is not thread-safe
+ In classic mode, constant autoloading is not thread-safe
```

2. Reword the sentence to reduce the ambiguity in it. The next sentence on line 324 is the same as this.

```diff
- In classic mode constant autoloading is not thread-safe
+ Constant autoloading is not thread-safe in classic mode
```